src/ucs/sys/ucs_netlink.c

```diff
@@ -0,0 +1,96 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "ucs_netlink.h"
+
+#include <ucs/sys/sock.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/type/status.h>
+#include <ucs/debug/log.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+
+static ucs_status_t ucs_netlink_socket_create(int *fd, int protocol)
+{
+    ucs_status_t ret;
+    struct sockaddr_nl sa = {0};
+
+    ret = ucs_socket_create(AF_NETLINK, SOCK_RAW, protocol, fd);
+    if (ret != UCS_OK) {
+        ucs_diag("failed to create netlink socket %d", ret);
+        *fd = -1;
+        return ret;
+    }
+
+    sa.nl_family = AF_NETLINK;
+
+    if (bind(*fd, (struct sockaddr*)&sa, sizeof(sa)) < 0) {
+        ucs_close_fd(fd);
+        ucs_diag("failed to bind netlink socket %d", *fd);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+ucs_status_t
+ucs_netlink_send_cmd(int protocol, void *nl_protocol_hdr,
```

Comment:
add `const` where possible

Reply:
`iovec::iov_base` is of type `void *` so can't use const. `recv_msg_buf` cannot be const as well